### PR TITLE
feat(model): Add linearsvc agent

### PR DESCRIPTION
**Pull request by its-sushant**
## Description
Added a agent for [Linear Support vector machine](https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html) model.

## Files

- **agents/linearsvc.py** (Python file to run linearsvc model)

## How to use
- Running **linearsvc** agent
    `atarashi -a linearsvc /path/to/file.c`

- Checking accuracy using **evaluator.py**
    `python3 path/to/evaluator.py -a linearsvc`

## Accuracy
The accuracy score that has been tested using [evaluator.py](https://github.com/fossology/atarashi/blob/master/atarashi/evaluator/evaluator.py) is **63%**

CC: @hastagAB @Kaushl2208 @GMishx @ag4ums @vasudevmaduri

### CODE REVIEW

**Review by hastagAB — COMMENTED** [review]
The agent looks good to me!
Need testing @Kaushl2208

**Review by Kaushl2208 — APPROVED** [review]
Chanegs Looks good to me :)

Will Test 👍

**Comment by amanjain97** [comment]
@hastagAB I think all PRs should have sphinx updates in them included or have we added this to github workflows ?

**Comment by Kaushl2208** [comment]
Tested the PR, Changes seems to be working :)

image


A small thing that we can look at a warning that popped up due to the change in version for unpickling the file with different versions:

image


You can read about it from here: https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations


Comments? @its-sushant @hastagAB @GMishx

**Comment by Kaushl2208** [comment]
> Tested the PR, Changes seems to be working :)
> 
> 
> A small thing that we can look at a warning that popped up due to the change in version for unpickling the file with different versions:
> 
> 
> You can read about it from here: https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
> 
> Comments? @its-sushant @hastagAB @GMishx

Is it happening because of old version of `linearsvc` being used in `linearsvc>=0.1.1` ??

Please try fixing it to a working version using `linearsvc==0.1.1` so pickling and unpickling does not cause issues.

**Comment by its-sushant** [comment]
> > Tested the PR, Changes seems to be working :)
> > A small thing that we can look at a warning that popped up due to the change in version for unpickling the file with different versions:
> > You can read about it from here: https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
> > Comments? @its-sushant @hastagAB @GMishx
> 
> Is it happening because of old version of `linearsvc` being used in `linearsvc>=0.1.1` ??
> 
> Please try fixing it to a working version using `linearsvc==0.1.1` so pickling and unpickling does not cause issues.

@Kaushl2208  Actually the model has been trained on `sklearn==1.0.2` and atarashi uses the updated version, that's why it is showing some warning. I need to train the model again with `sklearn==1.1.1` then I guess the warning will be gone.
